sbi/inference/abc/abc_base.py

```diff
@@ -87,3 +93,57 @@ def distance_fun(observed_data: Tensor, simulated_data: Tensor) -> Tensor:
             return distance(observed_data, simulated_data)
 
         return distance_fun
+
+    @staticmethod
+    def get_sass_transform(theta, x, expansion_degree=1, sample_weight=None):
+        """Return semi-automatic summary statitics function.
+
+        Running weighted linear regressin as in 
+        Fearnhead & Prandle 2012: https://arxiv.org/abs/1004.1112
+        
+        Following implementation in 
+        https://abcpy.readthedocs.io/en/latest/_modules/abcpy/statistics.html#Identity
+        and
+        https://pythonhosted.org/abcpy/_modules/abcpy/summaryselections.html#Semiautomatic
+        """
+        expansion = PolynomialFeatures(degree=expansion_degree, include_bias=False)
+        # Transform x, remove intercept.
+        x_expanded = expansion.fit_transform(x)
+        sumstats_map = np.zeros((x_expanded.shape[1], theta.shape[1]))
+
+        for parameter_idx in range(theta.shape[1]):
+            regression_model = LinearRegression(fit_intercept=True)
+            regression_model.fit(
+                X=x_expanded, y=theta[:, parameter_idx], sample_weight=sample_weight
+            )
+            sumstats_map[:, parameter_idx] = regression_model.coef_
+
+        sumstats_map = torch.tensor(sumstats_map, dtype=torch.float32)
+
+        def sumstats_transform(x):
+            x_expanded = torch.tensor(expansion.fit_transform(x), dtype=torch.float32)
+            return x_expanded.mm(sumstats_map)
+
+        return sumstats_transform
+
+    @staticmethod
+    def run_lra(
+        theta: torch.Tensor,
+        x: torch.Tensor,
+        observation: torch.Tensor,
+        sample_weight=None,
+    ):
+        """Return LRA adjusted parameters."""
```

Comment:
Would be good to spell out LRA and add a link to a reference, e.g, https://arxiv.org/abs/1707.01254